### PR TITLE
Revert "Shard-centric merge strategy (#1369)"

### PR DESCRIPTION
**Pull request by vangheem**
This reverts commit e2de02e971fb4bba89e3768d0ef8fb80b95c6de7.

### Description
We can continue working on this in a separate PR and make sure we have everything tested further

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

**Review by lferran — APPROVED** [review]

File: nucliadb/nucliadb/tests/integration/search/test_search.py [hunk 1]
```diff
@@ -1347,11 +1347,11 @@ async def test_search_two_logic_shards(
     # Check that search returns the same results
     resp1 = await nucliadb_reader.post(
         f"/kb/{kbid1}/search",
-        json=dict(query="dummy", vector=V1, min_score=-1, with_duplicates=True),
+        json=dict(query="dummy", vector=V1, min_score=-1),
```
Comment (tarekziade) [inline nucliadb/nucliadb/tests/integration/search/test_search.py:1350]: IIRC the `with_duplicate` change was an unrelated fix

Comment (lferran) [reply nucliadb/nucliadb/tests/integration/search/test_search.py:1350]: yes, but AFAIK this other fix was included in the same patch. not sure how easy it is to cherry pick
